### PR TITLE
Load roommates into linelist

### PR DESCRIPTION
**Pull request by ljwoodley**
It's functional but there's three TODOs:

~~1. Only read case data within the last 5 minutes. At the moment it's only reading data where the case_infection_area is given~~

~~2. Handling duplicated housing data from redcap. I sent an email earlier about this. The data from sftp is fine but there's true duplicated data based off the fact that the checkin date is updated whenever we get a new file. there's many work arounds to this. We can handle it in either the upload or whenever we do a housing data read.~~

3. Add non-uf family members. This will be in another PR

### CODE REVIEW

File: load_roomates_into_linelist.R [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            token = Sys.getenv("CASE_TOKEN"),
+            project_title = Sys.getenv("CASE_PROJECT_TITLE"))
+
+          if (length(write_data_to_case) > 2) {
+
+            email_subject <- paste("Case line list entries created", "|",
+                                   Sys.getenv("INSTANCE"), "|", script_run_time)
+
+            message <- paste("write_data_to_case$records_affected_count",
+                             "line-list entries written to ",
```
Comment (pbchase) [inline load_roomates_into_linelist.R:127]:
```suggestion
            message <- paste(write_data_to_case$records_affected_count,
                             "line-list entries written to",
```

You didn't want to quote that variable and there was an extra space on the second line

Comment (pbchase) [reply load_roomates_into_linelist.R:127]: I'll send this in a commit with a couple other changes

**Comment by pbchase** [comment]
I fixed a couple problems. In the process I created slightly different test code. this is what I used:

```
library(tidyverse)
library(dotenv)
library(REDCapR)
library(lubridate)
library(sendmailR)

source("R/data_fetching.R")
source("R/data_writing.R")

script_name <- "load_roommates_into_linelist"
script_run_time <- set_script_runtime()

# load_dot_env("prod.env")
dotenv::load_dot_env("roommate_test.env")

Sys.getenv("INSTANCE")

read_case_data <- read_redcap_db(
  redcap_uri = Sys.getenv("URI"),
  token = Sys.getenv("CASE_TOKEN"),
  project_title = Sys.getenv("CASE_PROJECT_TITLE"),
)

case_data_read <- read_case_data$data

read_housing_data <-  read_redcap_db(
  redcap_uri = Sys.getenv("URI"),
  token = Sys.getenv("HOUSING_TOKEN"),
  project_title = Sys.getenv("HOUSING_PROJECT_TITLE"),
  batch_size = 1000
)

housing_data_csv <- read_housing_data$data

ufid_in_housing_and_case <- housing_data_csv %>%
  distinct(ufid, infection_area) %>%
  drop_na() %>%
  filter(ufid %in% case_data$case_emp_ufid)

# copy infection area to cases
slice_of_case_data <- case_data_read %>%
  select(record_id, case_emp_ufid) %>%
  filter(!is.na(case_emp_ufid)) %>%
  left_join(ufid_in_housing_and_case %>%
              select(ufid, infection_area),
            by = c("case_emp_ufid" = "ufid")) %>%
  filter(!is.na(infection_area)) %>%
  arrange(infection_area) %>%
  slice(1:4) %>%
  rename("case_infection_area" = infection_area)
``` 

I run this instead of lines 1-22 of `load_roomates_into_linelist.R`, then I run line 26, then I change line 30 to read

```
    case_record_and_infection_area <- slice_of_case_data %>%
      distinct(record_id, case_infection_area)
```

Then I run line 30 and the rest of code.

I do not comment out the filters you had discussed.

**Comment by pbchase** [comment]
Also, I used a copy of Cases as my test project: https://redcap-phone.ctsi.ufl.edu/prod/redcap_v10.0.1/index.php?pid=47

File: load_roomates_into_linelist.R [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    if (length(read_housing_data) > 2) {
+
+      housing_data <- read_housing_data$data
+
+      # focus only on subjects with a ufid. Will deal with family members later
+      case_roomate_linelist_entry <- housing_data %>%
+        # remove non uf records
+        filter(!is.na(ufid)) %>%
+        # only students that are currently checked in
+        filter(checkin_indicator == 1) %>%
```
Comment (ljwoodley) [inline load_roomates_into_linelist.R:58]: This is set to only students that are currently checked in. We don't want to contact students that are not checked in right.

Comment (pbchase) [reply load_roomates_into_linelist.R:58]: For now, let's do that. The complexity of people-who-are-about-to-checkin can wait for another day it at all.

**Comment by pbchase** [comment]
That last commit looks good. It looks like you might be ready for a merge. Do you want me to merge?

**Comment by ljwoodley** [comment]
Yep, there's no more commits from me.

**Pull request merged**